source/IAccessibleHandler.py

```diff
@@ -781,7 +784,7 @@ def processDestroyWinEvent(window,objectID,childID):
 	# so can't use generic focus correction. (#2695)
 	focus=api.getFocusObject()
 	from NVDAObjects.IAccessible.mscandui import BaseCandidateItem
-	if objectID==0 and childID==0 and isinstance(focus,BaseCandidateItem) and window==focus.windowHandle and not eventHandler.isPendingEvents("gainFocus"):
+	if objectID == 0 and childID==0 and isinstance(focus,BaseCandidateItem) and not eventHandler.isPendingEvents("gainFocus"):

```

Comment:
You are now matching any destroy event with an objectID of 0 and childID of 0. This is not good.  It needs to be somehow limited to the candidate window.
Previously  it checked if it was the same window. but now you are saying it is different? In that case, what window is it exactly? I.e. is it a descendant of the focus window, or perhaps it has a specific window class?
I think it is more likely that the modern candidate UI is simply not firing a destroy event at all, and perhaps your code is matching on some other random one?

Reply:
I found a specific window class  CiceroUIWndFrame